app/Repositories/EpochRepository.php

```diff
@@ -49,14 +49,15 @@ public function endEpoch($circle_id)
                     $tokenGift->save();
                 }
 
-                $this->model->where('circle_id', $circle_id)->delete();
-                $users = User::where('circle_id', $circle_id)->get();
-                foreach ($users as $user) {
-                    $user->histories()->create(['bio' => $user->bio, 'epoch_id' => $epoch->id, 'circle_id' => $circle_id]);
+                $this->model->where('circle_id',$circle_id)->delete();
+                $users = User::where('circle_id',$circle_id)->get();
+                foreach($users as $user) {
+                    $user->histories()->create(['bio' => $user->bio,'epoch_id' => $epoch->id, 'circle_id' => $circle_id]);
+                    $user->give_token_remaining = $user->starting_tokens;
+                    $user->save();
                 }
-                User::where('circle_id', $circle_id)->where('non_giver', 0)->yetToSend()->update(['non_receiver' => 1]);
-                User::where('circle_id', $circle_id)->update(['bio' => null, 'give_token_received' => 0, 'give_token_remaining' => DB::raw("`starting_tokens`"), 'epoch_first_visit' => 1]);
-
+                User::where('circle_id',$circle_id)->where('non_giver',0)->yetToSend()->update(['non_receiver'=>1]);
+                User::where('circle_id',$circle_id)->update(['bio'=> null,'give_token_received'=>0, 'epoch_first_visit' => 1]);
```

Comment:
missing the change to `give_token_remaining`?

Reply:
I moved it to line 56-57, to avoid using the DB::raw
```
                    $user->give_token_remaining = $user->starting_tokens;
                    $user->save();
```